docs/Installation/InstallationOnAppleSilicon.md

```diff
@@ -0,0 +1,163 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+## Installing SpECTRE on Apple Silicon {#installation_on_apple_silicon}
+
+The following instructions show how to install SpECTRE and its dependencies
+on Apple Silicon Macs. Apple Silicon is an arm64 architecture, in contrast
+to the x86-64 architecture that SpECTRE usually targets. These instructions
+will result in an Apple Silicon native build of SpECTRE.
+
+\note Floating-point exception trapping and python bindings are not currently
+supported on Apple Silicon.
+
+### 0. Install the xcode command-line tools.
+
+Install the xcode command-line tools, which include the clang compiler, etc.
+
+~~~
+xcode-select --install
+~~~
+
+### 1. Make a directory to install prerequisites
+
+First, make a directory to hold some prerequisites that spectre depends on.
+Name this directory whatever you like, e.g. `$HOME/apps`.
+~~~
+export SPECTRE_DEPS_ROOT=$HOME/apps
+mkdir $SPECTRE_DEPS_ROOT
+cd $SPECTRE_DEPS_ROOT
+mkdir src
+cd src
+~~~
+
+### 2. Install python dependencies
+
+Spectre depends on python and some python packages. Miniforge is a way to
+install an arm64-native python stack on Apple Silicon Macs. Download
+the [Miniforge3-MacOSX-arm64.sh](https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh)
+installation script, and then do the following:
+~~~
+# Install miniforge. Accept all default choices
+bash Miniforge3-MacOSX-arm64.sh
+# Activate conda base environment for current session
+eval "$(/Users/gwpacworkstation/miniforge3/bin/conda shell.zsh hook)"
+# Activate conda at startup (to make sure you're always using native python)
+conda init zsh
+exit # close shell. conda will automatically load when new zsh shells start.
+~~~
+
+Open a new terminal window. Then, install the necessary python packages. (Note:
+jupyter and notebook are recommended, not required.)
+~~~
+conda install numpy scipy matplotlib h5py pyyaml jupyter notebook
+~~~
+
+### 3. Install dependencies with Homebrew
+
+Most of spectre's dependencies beyond python can be installed using the
+[homebrew](https://brew.sh) package manager. First, install Homebrew by
+following the instructions on the [homebrew](https://brew.sh) homepage. Then,
+run the following to install a fortran compiler and other dependencies:
+~~~
+brew install gcc
+brew install openblas boost gsl cmake doxygen
+brew install ccache autoconf automake catch2 jemalloc hdf5 pybind11 yaml-cpp
+~~~
+
+### 4. Install remaining dependencies
+
+Here, install the remaining dependencies that cannot be installed
+with homebrew or miniforge.
```

Comment:
I think that ideally we would merge this whole page into `Installation.md`, since it will surely become out-of-date rather soon. We don't have to repeat explanations on how to clone spectre, how to build charm, how to create build directories, how to run cmake, etc. I restructured the installation page a bit in #3712 to accommodate this better.

Then, the remaining ARM-specific instructions could be added as notes to the installation page, if they are not already taken care of by the corresponding Spack packages. For instance, the `-march=native` issue with libsharp is already handled automatically by this Spack patch: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/libsharp/1.0.0-arm.patch. Therefore, I think the Spack instructions in #3712 might already be sufficient to install the dependency tree on Apple Silicon, though I haven't tested it of course. Could you perhaps give it a try @geoffrey4444?

This page is of course a lot more hands-on tutorial-style than the `Installation.md` page (even with #3712). I suggest that instead of maintaining these step-by-step installation tutorials for a specific system we add enough details to the `Installation.md` page so new people can follow it. We already have Docker, Singularity and Spack instructions that are reasonably system-independent. Manual installation instructions depend so much on the specific system that I don't think we can maintain them. It's each dependency's job to give installation instructions, and we already link to them. What do you think @geoffrey4444?

Reply:
I thought these instructions would be useful to have around in this level of detail primarily for the Fullerton undergrads, since we have a bunch of M1 Macs. I'd be willing to support it, but if you think a separate page isn't a good idea, I'm fine closing this PR

Reply:
spack cannot yet build charm++ for a M1 Mac...

Reply:
nor libxsmm...

Reply:
Have you had any success with the instructions as posted in this PR @kidder ?

Reply:
the only thing I was able to install with spack was libsharp; had to install blaze and brigand by hand as well

Reply:
ok let's add this page then, though I think it would be nice to (1) contribute fixes for ARM builds upstream to Spack and (2) merge these detailed instructions into the main `Installation.md` page.